### ISSUE TITLE
Much better documentation about what we're building and why

### ISSUE THREAD

**Issue by bketelsen** [gopheracademy/congo#10]
Fill out the wiki -- what are we building? Why are we building it?  What is the goal?  What are the pieces? What will it do?  Who will use it?

**Comment by jamra** [comment]
Can we start with what are the needs of a conference? Maybe you can use the research you did with the other tools out there to come up with the core requirements.

**Comment by bketelsen** [comment]
@jamra did you see the wiki?  I started with a pretty comprehensive list of things we need there.  Does it need more?

**Comment by jamra** [comment]
I'll take a look

**Comment by jamra** [comment]
It's a start, but I'm thinking we could do with some use case diagrams. Nothing too serious, just a breakdown of the types of users.

So far, we have a todo list that sends emails and has spreadsheets. I'd like to get some more details.

For example, a user logs in and goes to a vendors view (private to highest level of admins).The user then can <insert use case here>. It helps me to know what you need to do with the vendor. What is actionable? Is the two way communication handled by email or is it monitored by this system?

Alternatively, Customers can vote on certain things like Speakers. Additionally, Tasks can be assigned to Users and Attendees can purchase tickets. This gives us an idea that can lead into UI.

**Comment by bketelsen** [comment]
excellent.  I'll work on this sort of documentation, and hopefully convince @erikstmartin to help too

**Comment by jamra** [comment]
That would be great. Once you're done, I wouldn't mind creating some mockups with Balsamiq. That's usually a good way to get feedback from the community and make sure that the features we are implementing don't end up sparse and disconnected.

**Comment by bketelsen** [comment]
https://github.com/gopheracademy/congo/wiki/Constituents
what about something like this?

**Comment by jamra** [comment]
That looks good. From Asana's site, it seems that you can create general posts and assign tasks based on those posts. Who in the Constituent list would a task be assigned to and who does the assigning?

**Comment by bketelsen** [comment]
I don't know the answer to that yet, but I think that many different constituents should be able to create tasks and assign them to others.  Examples:
Event Planner creates a task to make a new logo assigns to Web Developer
Sponsor creates a task to source giveaways and assigns to someone on the sponsor team

Some tasks may be system-created.  Un-confirmed sponsors should generate a follow-up task to an event planner after X days

**Comment by erikstmartin** [comment]
Agreed, there is a couple of things here. So there is the notion of a ticket being "assigned" to a person, they are the person responsible for it. There is also the notion of "linking" it to another resource. A sponsor, a speaker, a vendor, etc. 

This is mainly so when you're working with a speaker/sponsor/vendor via email or phone you don't have to keep reaching out for one thing at a time, you can see all tasks associated with that person and address them at once.
